File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/ui/toolwindow/pullrequestlist/columns/IssuesColumnInfo.java
package com.duxtinto.intellij.plugin.github.codereviews.ui.toolwindow.pullrequestlist.columns;

import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.PullRequestEntity;
import com.intellij.util.ui.ColumnInfo;
import org.jetbrains.annotations.Nullable;

import java.util.stream.Collectors;

public class IssuesColumnInfo extends ColumnInfo<PullRequestEntity, String> {
    public IssuesColumnInfo() {
        super("issues");
    }

    @Nullable
    @Override
    public String valueOf(PullRequestEntity pullRequest) {
        return pullRequest.getCloseableIssues()
                .stream()
                .map(issue -> String.valueOf(issue.getNumber()))
                .collect(Collectors.joining(","));
    }
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/ui/toolwindow/pullrequestlist/columns/IssuesColumnInfo.kt
package com.duxtinto.intellij.plugin.github.codereviews.ui.toolwindow.pullrequestlist.columns

import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.PullRequestEntity
import com.intellij.util.ui.ColumnInfo

import java.util.stream.Collectors

class IssuesColumnInfo : ColumnInfo<PullRequestEntity, String>("issues") {

    override fun valueOf(pullRequest: PullRequestEntity): String? {
        return pullRequest
                .closeableIssues
                .joinToString(",") {
                    (number) -> number.toString()
                }
    }
}
